life_hc/MySQL/Items/fn_spikeStrip.sqf

```diff
@@ -1,24 +1,47 @@
-#include "\life_hc\hc_macros.hpp"
-/*
-    File: fn_spikeStrip.sqf
-    Author: Bryan "Tonic" Boardwine
-
-    This file is for Nanou's HeadlessClient.
-
-    Description:
-    This is the server-side part of it which constantly monitors the spike strip and vehicles near it.
-    First originally tried triggers but I was never any good at those nor do I like them as they
-    have a global effect.
-*/
-private ["_nearVehicles","_spikeStrip"];
-_spikeStrip = [_this,0,objNull,[objNull]] call BIS_fnc_param;
-if (isNull _spikeStrip) exitWith {}; //Bad vehicle type passed.
-
-waitUntil {_nearVehicles = nearestObjects[getPos _spikeStrip,["Car"],5]; count _nearVehicles > 0 || isNull _spikeStrip};
-
-if (isNull _spikeStrip) exitWith {}; //It was picked up?
-_vehicle = _nearVehicles select 0;
-
-if (isNil "_vehicle") exitWith {deleteVehicle _spikeStrip;};
-[_vehicle] remoteExec ["life_fnc_spikeStripEffect",_vehicle];
-deleteVehicle _spikeStrip;
\ No newline at end of file
+#include "\life_hc\hc_macros.hpp"
+/*
+    File: fn_spikeStrip.sqf
+    Author: DomT602
+
+    This file is for Nanou's HeadlessClient.
+
+    Description:
+    Adds spikestrip to hc-side array and if required - starts the monitoring of spikestrips.
+*/
+params [
+    ["_spikeStrip", objNull, [objNull]]
+];
+if (isNull _spikeStrip) exitWith {};
+
+hc_spikes pushBack _spikeStrip;
+
+if (hc_spikes isEqualTo []) then { //start monitoring spikestrips
+    private _minSpikeSpeed = LIFE_SETTINGS(getNumber,"minimumSpikeSpeed");
+
+    for "_i" from 0 to 1 step 0 do {
+        if (hc_spikes isEqualTo []) exitWith {};
+        private _toDelete = [];
+
+        {
+            private _pos = getPosATL _x;
+            (nearestObjects [_pos,["Car"],5]) params [["_nearVeh", objNull]];
+            if (alive _nearVeh && {abs (speed _nearVeh) > _minSpikeSpeed}) then {
+                [_nearVeh] remoteExecCall ["life_fnc_spikeStripEffect",_nearVeh];
+                deleteVehicle _x;
+                _toDelete pushBack _forEachIndex;
+            } else {
+                if (isNull _object) then {
+                    deleteVehicle _x;
+                    _toDelete pushBack _forEachIndex;
+                };
+            };
+        } forEach hc_spikes;
+
+        reverse _toDelete;

```

Comment:
why reverse? shouldn't make a difference

Reply:
You have to delete backwards as if I delete index 0, the hc_spikes array has a new index and so forth, hence why deleting from it backwards never affects earlier indices

Reply:
you got a point there :D

One more thought...
Is saving an array of indexes and iterating over them to `deleteAt` faster?

since you delete the objects, they become `objNull` and a simple `hc_spikes = hc_spikes - [objNull]` should do the same without iterating again